packages/common/src/services/coingecko/CoingeckoClient.ts

```diff
@@ -49,33 +49,73 @@ export class CoingeckoClient {
     vs_currency: string,
     from: UnixTime,
     to: UnixTime,
+    address?: EthereumAddress,
   ): Promise<CoinMarketChartRangeData> {
-    const data = await this.query(
-      `/coins/${coinId.toString()}/market_chart/range`,
-      {
-        vs_currency: vs_currency.toLowerCase(),
-        from: from.toString(),
-        to: to.toString(),
-      },
-    )
-
-    const parsedData = CoinMarketChartRangeResult.parse(data)
+    try {
```

Comment:
I like the idea, dislike the implementation. Ideally this function is super simple, like 3 lines: 1. call `market_chart/range` 2. if error get new coingeckoId 3. call `market_chart/range` with new id

Steps 1 and 3 get a new function, step 2 gets a new function.

The function for getting new coingecko id should ideally cache the result and only re-fetch every hour (configurable).